src/Fantomas/SourceParser.fs

```diff
@@ -800,17 +800,17 @@ let (|App|_|) e =
 let (|AppTuple|_|) =
     function
     | App (SynExpr.DotGet _, [ (Paren (_, Tuple _, _)) ]) -> None
-    | App (e, [ (Paren (lpr, Tuple args, rpr)) ]) -> Some(e, lpr, args, rpr)
+    | App (e, [ (Paren (lpr, Tuple (args, rangeOfTuple), rpr)) ]) -> Some(e, lpr, args, rangeOfTuple, rpr)
     | App (e, [ (Paren (lpr, singleExpr, rpr)) ]) ->
         match singleExpr with
         | SynExpr.Lambda _
         | SynExpr.MatchLambda _ -> None
-        | _ -> Some(e, lpr, [ singleExpr ], rpr)
+        | _ -> Some(e, lpr, [ singleExpr ], range.Zero, rpr)
```

Comment:
Please use an option type here instead.
This will better indicate that the range of the tuple is not always there.